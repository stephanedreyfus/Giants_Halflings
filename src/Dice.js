import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Die from './Die';
import { Dice } from './styling/DiceStyle';

class Dice extends Component {
  render() {
    return (
      <Dice>
        {this.props.dice.map((d, idx) =>
          <Die
            handleClick={this.props.handleClick}
            val={d}
            locked={this.props.locked[idx]}
            idx={idx}
            key={idx}
          />
        )}
      </Dice>
    );
  }
}

Dice.propTypes={
  dice: PropTypes.array,
  locked: PropTypes.array,
  handleClick: PropTypes.func.isRequired,
}

export default Dice;