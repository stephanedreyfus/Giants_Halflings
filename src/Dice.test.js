import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Dice from './Dice';

const myMock = jest.fn();

it('renders without crashing', () => {
  shallow(
    <Dice
    dice={[1, 2, 6, 4, 4]}
    locked={[false, false, false, true, true]}
    hadleClick={myMock}
    />
  );
});

it('matches snapshot', () => {
  const wrapper = shallow(
    <Dice
    dice={[1, 2, 6, 4, 4]}
    locked={[false, false, false, true, true]}
    hadleClick={myMock}
    />
  );
  const serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
