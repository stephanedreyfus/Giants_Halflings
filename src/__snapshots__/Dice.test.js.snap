// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<styled.div>
  <Die
    handleClick={[Function]}
    idx={0}
    key="0"
    locked={false}
    val={1}
  />
  <Die
    handleClick={[Function]}
    idx={1}
    key="1"
    locked={false}
    val={2}
  />
  <Die
    handleClick={[Function]}
    idx={2}
    key="2"
    locked={false}
    val={6}
  />
  <Die
    handleClick={[Function]}
    idx={3}
    key="3"
    locked={true}
    val={4}
  />
  <Die
    handleClick={[Function]}
    idx={4}
    key="4"
    locked={true}
    val={4}
  />
</styled.div>
`;
